workbench-core/environments/src/services/environmentTypeService.ts

```diff
@@ -49,20 +57,43 @@ export default class EnvironmentTypeService {
 
   /**
    * List environment type objects from DDB
-   * @param pageSize - the number of environment type objects to get (optional)
-   * @param paginationToken - the token from the previous page for continuation (optional)
+   * @param request - pagination, filter and sorting parameters
    *
    * @returns environment type objects
    */
   public async listEnvironmentTypes(
-    pageSize?: number,
-    paginationToken?: string
-  ): Promise<{ data: EnvironmentType[]; paginationToken: string | undefined }> {
+    request: ListEnvironmentTypeRequest
+  ): Promise<{ data: EnvironmentType[]; paginationToken?: string }> {
+    const { filter, sort, pageSize, paginationToken } = request;
+    validateSingleSortAndFilter(filter, sort);
     let queryParams: QueryParams = {
       key: { name: 'resourceType', value: this._resourceType },
       index: 'getResourceByCreatedAt',
       limit: pageSize && pageSize >= 0 ? pageSize : DEFAULT_API_PAGE_SIZE
     };
+    const gsiPropertyNames = ['Name', 'Status'];
+    if (filter) {
+      const filterProperty = Object.keys(filter)[0] as keyof EnvironmentTypeFilter;
+      const gsiFilterProperty = gsiPropertyNames.filter((prop) => prop.toLowerCase() === filterProperty)[0];
+      //eslint-disable-next-line security/detect-object-injection
+      const queryParamsFilter = parseQueryParamFilter(
+        filter[filterProperty],
+        filterProperty,
+        `getResourceBy${gsiFilterProperty}`
+      );
+      queryParams = { ...queryParams, ...queryParamsFilter };
+    }
+    if (sort) {
+      const sortProperty = Object.keys(sort)[0] as keyof EnvironmentTypeSort;
+      const gsiSortProperty = gsiPropertyNames.filter((prop) => prop.toLowerCase() === sortProperty)[0];
+      //eslint-disable-next-line security/detect-object-injection
```

Comment:
Can we do this without enabling [Object Injection](https://github.com/nodesecurity/eslint-plugin-security/blob/main/docs/the-dangers-of-square-bracket-notation.md)?

Reply:
fixed in next commit